src/testFixtures/kotlin/com/exactpro/th2/common/TestUtils.kt

```diff
@@ -53,26 +60,81 @@ fun assertEqualBatches(expected: MessageBatch, actual: MessageBatch, lazyMessage
     }
 }
 
-fun assertEqualMessages(expected: Message, actual: Message, lazyMessage: () -> String? = {null}) {
+fun assertEqualGroups(expected: MessageGroupOrBuilder, actual: MessageGroupOrBuilder, lazyMessage: () -> String? = {null}) {
+    Assertions.assertEquals(expected.messagesCount, actual.messagesCount) {"wrong count of messages in group: \n${TextFormat.shortDebugString(actual)}"}
+    expected.messagesList.forEachIndexed { i, message ->
+        try {
+            assertEqualMessages(message, actual.messagesList[i], lazyMessage)
+        } catch (e: AssertionFailedError) {
+            throw AssertionFailedError(
+                "Error in message from group with index '$i'.\n${e.message}",
+                e.expected,
+                e.actual,
+                e.cause
+            )
+        }
+    }
+}
+
+fun assertEqualGroupBatches(expected: MessageGroupBatchOrBuilder, actual: MessageGroupBatchOrBuilder, lazyMessage: () -> String? = {null}) {
+    Assertions.assertEquals(expected.groupsCount, actual.groupsCount) {"wrong count of groups in batch: \n${TextFormat.shortDebugString(actual)}"}
```

Comment:
> TextFormat.shortDebugString

I'd suggest using JSON instead - it can be formatted afterwards for easier investigation